Design.md

```diff
@@ -181,7 +181,9 @@ This API creates a user on Vault.
 1. The `get-policy` API will be called with the `userPolicy@[account-id]` policy name using assumed role credentials
 1. If the `get-policy` API does not return any policy, using the assumed role credentials creates an IAM managed policy with the `userPolicy@[account-id]` policy name with full S3 access.
 1. `attach-policy` for the new user will be called using the assumed role credentials and the policy arn formatted as `arn:aws:iam::[account-id]:policy/userPolicy@[account-id]`.
-1. `create-s3credentials` api will be called using assumed role credentials
+1. IAM `generate-access-key` api will be called using assumed role credentials
+1. Encrypt the `secret-key` using AES algorithm and store the encrypted `secret-key` on Redis Cache in the hash named as `OSIS` with the hash-key formatted as `<Username>__<AccessKeyID>`. 
```

Comment:
Per discussion, it should be in base 64 right?
As there will be multiple keys stored, I would suggest checking out Utapi, for the format.

Reply:
Also worth mentioning redis-sentinel.

Reply:
This is changed after discussing with Vianney over Slack. We decided that AES is more secure when stored on Redis than just base64 encoding. AES private key will be provided on `application.properties` for OSIS and should be common for all nodes running OSIS.

Reply:
Makes sense :) 
Just wanted to be sure.